compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -4326,9 +4326,7 @@ static void unsignedIntCompareNarrower(TR::Node * node, TR::Simplifier * s, TR::
       TR::ILOpCodes secondOp = secondChild->getOpCodeValue();
       if (firstOp == TR::su2i && firstChild->getReferenceCount()==1)
          {
-         if (secondOp == TR::su2i ||
-             (secondOp == TR::iuconst               &&
-              secondChild->getUnsignedInt() <= USHRT_MAX))
+         if (secondOp == TR::su2i || secondChild->getUnsignedInt() <= USHRT_MAX)
```

Comment:
Is this logically equivalent to what was there before the change ? It seems to me that you should have omitted the ```secondChild->getUnsignedInt() <= USHRT_MAX``` test on the basis that ```secondOp == TR::iuconst``` should be false.

Reply:
Yes, this was a mistake. `(secondOp == TR::iuconst&&secondChild->getUnsignedInt() <= USHRT_MAX)` should all be omitted as it will always be `false`. Fix in the new force-push.